client/src/App.jsx

```diff
@@ -1,36 +1,41 @@
 import { useEffect, useState } from "react";
 import "bootstrap/dist/css/bootstrap.min.css";
 import "./App.css";
+import { jwtDecode } from "jwt-decode";
 import EmbeddedPlayer from "./components/EmbeddedPlayer";
 import AddSongForm from "./components/AddSongForm";
 import Header from "./components/header/Header";
 import Background from "./components/Background";
+import UpdateSpotifyPlaylistButton from "./components/UpdateSpotifyPlaylistButton";
 
 function App() {
-  //getting token from query string sent from backend
-  const search = window.location.search;
-  const params = new URLSearchParams(search);
-  const maciToken = params.get("token");
-  // check if query return equals null and if it does not, set sessionstorage to maciToken
-  if (maciToken !== null) {
-    //set session storage data for maciToken
-    sessionStorage.setItem("maciToken", maciToken);
-    console.log(`sessionStorageData: ${sessionStorage.getItem("maciToken")}`);
-  }
+  const [userRole, setUserRole] = useState(null);
 
-  const [isLoggedIn, setLogin] = useState(false);
   useEffect(() => {
-    if (sessionStorage.getItem("maciToken") !== null) setLogin(true);
+    //getting token from query string sent from backend
+    const search = window.location.search;
+    const params = new URLSearchParams(search);
+    const maciToken = params.get("token");
+    // check if query return equals null and if it does not, set sessionstorage to maciToken
+    if (maciToken !== null) {
```

Comment:
Suggestion: With the code review we had yesterday, I think it'd be worth checking for undefined too as we've learned they are two different things :)